### PR TITLE
Update Transition Networks support

### PR DESCRIPTION
**Pull request by justinh-rahb**
Sorry, I do not know git at all, still learning. My development environment was not properly setup, I've since re-read the guides in your very helpful docs. Mea culpa. Anyway...

SNMP on these is a bit weird, sometimes it outputs a lot of useful data, other times it seems to be lacking even basic interface statistics. Reboots sometimes help.

At best, this addition should output a hardware ID, if not it remains blank as it always was. Tested against Transition SM8TAT2SA 8-port managed PoE+ switch.

DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [x] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

**Comment by justinh-rahb** [comment]
OK after further reviewing my work against another test article with a new development environment, this needs some work. I know this information is buried somewhere in the SNMP output if you can get it working correctly. Will get to the bottom of it when I'm on the clock tomorrow.

File: includes/polling/os/transition.inc.php [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+ * Copyright (c) 2014 Neil Lathwood <https://github.com/laf/ http://www.lathwood.co.uk/fa>
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the
+ * Free Software Foundation, either version 3 of the License, or (at your
+ * option) any later version.  Please see LICENSE.txt at the top level of
+ * the source code distribution for details.
+ */
+
+$objectID = preg_replace('/[\r\n\"]+/', ' ', snmp_get($device, ".1.0.8802.1.1.2.1.3.3.0", "-OQv"));
```
Comment (murrant) [inline includes/polling/os/transition.inc.php]: \r \n and " are already trimmed in the snmp_get function, so no need to do it again.

File: includes/polling/os/transition.inc.php [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+$data = $objectID;
```
Comment (murrant) [inline includes/polling/os/transition.inc.php]: this seem silly, just use one variable please.

Comment (justinh-rahb) [reply includes/polling/os/transition.inc.php]: Fixed.

File: includes/polling/os/transition.inc.php (hunk 1 above)
Comment (justinh-rahb) [reply includes/polling/os/transition.inc.php]: Fixed.

**Comment by justinh-rahb** [comment]
There is already existing test data for transition but it was done with the MIBs are active:
[https://github.com/librenms/librenms/pull/9729](url)

Please advise, would you like me to replace this test data?

**Comment by murrant** [comment]
Then that snmprec just needs 1.0.8802.1.1.2.1.3.3.0 added to it.

**Comment by justinh-rahb** [comment]
Fantastic, one PR closer to my daily updates working again! 

Always waiting on Travis, that guy is lazy :-P

**Comment by murrant** [comment]
This pull request has been mentioned on **LibreNMS Community**. There might be relevant details there:

https://community.librenms.org/t/v1-57-release-changelog-october-2019/9969/1

**Pull request merged**